### PR TITLE
Add More Json tests

### LINKED ISSUE #3939: Feature: Reuse existing dataset in json testing
### API

Other

### Description

We currently have a bunch of json files used for testing purpose, however a better design is to reuse existing csv files.
Similar to parquet test, we should implement a function convertcsvtojson() function which converts a csv dataset to a json dataset.

### PR DESCRIPTION
**Pull request by mxwli**
# Description

Reuses csvtoparquet and renames.

Fixes #3939

# Contributor agreement

- [x] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -171,10 +171,21 @@ extension-test-build:
 		-DENABLE_ADDRESS_SANITIZER=TRUE \
 	)
 
+extension-json-test-build:
```
Comment (acquamarin) [inline Makefile:174]: Why not reuse the existing test command?
Do we need this extra command?

File: extension/json/test/tinysnb_all_shortest_path_json.test [hunk 2]
```diff
@@ -1,4 +1,4 @@
--DATASET CSV tinysnb_json
+-DATASET JSON tinysnb_json
```
Comment (acquamarin) [inline extension/json/test/tinysnb_all_shortest_path_json.test:1]: Remove the tinysnb json dataset, use the conversion function instead.

File: extension/test_list [hunk 3]
```diff
@@ -0,0 +1,57 @@
+httpfs~test~test_files~s3_glob.DISABLED_GlobS3Folder /data/maxwell/workspace/kuzu/kuzu/extension/httpfs/test/test_files/s3_glob.test
```
Comment (acquamarin) [inline extension/test_list:1]: remove

File: test/include/test_runner/csv_converter.h [hunk 4]
```diff
@@ -7,30 +7,30 @@
 namespace kuzu {
 namespace testing {
 
-// Convert an entire CSV dataset directory to parquet.
+// Convert an entire CSV dataset directory to some specified output.
```
Comment (acquamarin) [inline test/include/test_runner/csv_converter.h:10]: I would suggest moving the csv->json class to a separate header and source files.
And have a base common class CSVConverter.

File: extension/json/test/tinysnb_all_shortest_path_json.test (hunk 2 above)
Comment (mxwli) [reply extension/json/test/tinysnb_all_shortest_path_json.test:1]: I wouldn't do this. `tinysnb_json` covers cases like unstructured or missing fields, which isn't covered by conversion.

File: test/include/test_runner/csv_converter.h (hunk 4 above)
Comment (mxwli) [reply test/include/test_runner/csv_converter.h:10]: csv->parquet and csv->json aren't different classes. The only part of the code that actually defines the class to convert to `parquet` is a `".parquet"` file extension added on top of export destinations. The only real change here is that I changed the hardcoded `".parquet"` to an `outputFileExtension` variable.

File: Makefile (hunk 1 above)
Comment (mxwli) [reply Makefile:174]: The existing test command requires that people working on JSON also build DuckDB.

File: test/runner/e2e_test.cpp [hunk 5]
```diff
@@ -53,22 +53,34 @@ class EndToEndTest : public DBTest {
...
+        default: {
             dataset = TestHelper::appendKuzuRootPath("dataset/" + dataset);
         }
+        }
     }
 
     void TearDown() override {
         std::filesystem::remove_all(databasePath);
         BaseGraphTest::removeIEDBPath();
         if (datasetType == TestGroup::DatasetType::CSV_TO_PARQUET) {
```
Comment (ray6080) [inline test/runner/e2e_test.cpp:80]: Can u merge these two cases into one? 
```
if (datasetType == TestGroup::DatasetType::CSV_TO_PARQUET || datasetType == TestGroup::DatasetType::CSV_TO_JSON)
```

File: test/test_runner/csv_converter.cpp [hunk 6]
```diff
@@ -118,27 +118,31 @@ void CSVToParquetConverter::readCopyCommandsFromCSVDataset() {
...
-    std::ofstream outfile(parquetCopyFile);
+    auto outputCopyFile =
+        LocalFileSystem::joinPath(outputDatasetPath, std::string(TestHelper::COPY_FILE_NAME));
+    std::ofstream outfile(outputCopyFile);
     if (!outfile.is_open()) {
         throw TestException(
-            stringFormat("Error opening file: {}, errno: {}.", parquetCopyFile, errno));
+            stringFormat("Error opening file: {}, errno: {}.", outputCopyFile, errno));
+    }
+    if (fileExtension == ".json") {
```
Comment (ray6080) [inline test/test_runner/csv_converter.cpp:135]: Do we need to load json extension here? I can see we load twice, here and also in `convertCSVFiles`.

**Review by ray6080 — APPROVED** [review]

File: test/test_runner/csv_converter.cpp (hunk 6 above)
Comment (mxwli) [reply test/test_runner/csv_converter.cpp:135]: Yes. One for csv->json conversions, one for json->internal importing once the tests run.

**Pull request merged**